About tizen 7.0 (#1737)
Samsung/tizen-docs: docs/platform/what-is-tizen/versions/tizen-7-0-m2.md

# Tizen 7.0

The second milestone (M2) release of Tizen 7.0 was announced in October 2022. 

> Tizen is an open-source operating system (OS) maintained by Samsung, it is developed and optimized for smart devices since 2012.

Reviewer: ```suggestion
Tizen is an open-source operating system (OS) maintained by Samsung, and it is being developed and optimized for smart devices since 2012.
```

The main progress of Tizen 7.0 includes work considering new Tizen devices and strengthening intelligent device and service platform features. Core platform technologies are needed to prepare for our future with the Metaverse and robots.

> These new enhancements include 3D rendering engine enhancement, real-time enabled kernel and more.

Reviewer: ```suggestion
These new enhancements include 3D rendering engine enhancement, real-time enabled kernel, and more.
```

## Natural User Interface

- **Mini Avatar UI Assistant(Consistent Interaction Through Voice)**:

  Tizen provides UI assistance technology that allows users to control the UI through voice without additional processing in applications

  The Mini avatar is displayed in a separate layer. And it enables more natural interaction by showing the appropriate feedback for the situation.
  And it shows automatically generated command for all controllable UI objects, and when user says a command, it sends events to a selected object.

  **Figure: Mini Avatar UI Assistant**

  ![MiniAvartarUIAssistance](./media/7.0_MiniAvartarUIAssistance.png)

- **3D User Interface**:

  Tizen provides a unified rendering engine. All UI elements have depth information and they are located in 3D space

  As all UI elements are rendered together, we can optimize the rendering pass and minimize memory usage.

	The biggest advantage is that it reduces the complexity quite significantly for creating a 3D UI.

> 	This is because almost all the features used in 2D UI applications such as event handling, text input, animations and accessibility etc., are still available in 3D

Reviewer: ```suggestion
	This is because almost all the features used in 2D UI applications such as event handling, text input, animations, and accessibility, etc., are still available in 3D
```

	**Figure: 3D Scene Navigation**

  ![3DSceneNavigation](./media/7.0_3DSceneNavigation.png)

## Platform Compatibility

- **Tizen-Android HAL Adaptation**:

  Lots of time and effort are used to develop hardware abstraction layers and Tizen has introduced HAL APIs in Tizen 6.5.
	While we had some success with this, many SoC chip makers only support Android. To address this issue,
	we introduced a new solution to run Tizen on Android supported SoC by providing an adaptation bridge to Android HAL
	
	To make Tizen run on Android HAL and Kernel, we added the open source project Libhybris and defined a new glue layer to access hardware devices through Android HAL.
  Also, to solve compatibility issues between bionic in Android and glibc in Tizen, we used separate namespaces for the Android System and Hal container.
  
  **Figure: Tizen-Android Hal Adaptation**

  ![TizenAndroidHalAdaptation](./media/7.0_TizenAndroidHalAdaptation.png)

## Cross Platform Technology
	
	If you are already using popular cross-platform frameworks such as Web, Flutter or Maui, you can deploy your services to Tizen with minimal effort
	
- **New support for Tizen Web eco platform**:

>   Web is arguably the most widely used cross-platform technology. And more and more services are based on web on Tizen devices also,

Reviewer: ```suggestion
  Web is arguably the most widely used cross-platform technology. And more and more services are based on web on Tizen devices,
```

>   so it is business critical to support up-to-date web engines.To achieve that, we had to reduce time and effort to port a new version of the open source Chromium web engine.

Reviewer: ```suggestion
  so it is business critical to support up-to-date web engines. To achieve that, we had to reduce time and effort to port a new version of the open source Chromium web engine.
```

  A big move we took with Tizen 7.0 was re-architecturing our web engine porting layer. Instead of using our thick Tizen platform port,

>   we now use Chromium’s aura windows system and ozone abstraction layer.As a result, it reduced significant porting effort, around 50% of code was removed,

Reviewer: ```suggestion
  we now use Chromium’s aura windows system and ozone abstraction layer. As a result, it reduced significant porting effort, around 50% of code was removed,
```

  as in LOC, more than 100,000 lines total.

  **Figure: New support for Tizen Web eco platform**

  ![NewSupportForTizenWebEcoPlatform](./media/7.0_NewSupportForTizenWebEcoPlatform.png)

- **Flutter Support Improvement**:

  Flutter was introduced in 2016 and has gained huge popularity among developers with more than 600,000 apps in the Google Play Store today.
  Last year, we introduced Flutter Tizen SDK at here SDC 2021, and in Tizen 7.0, we continue to support more and more first party plugins such as Google Maps,
  and third-party plugins such as audio players, and also Tizen dedicated Flutter plugins for Tizen specific feature access. Also, you can now use Google sign-in on Flutter Tizen TVs apps.

>   New accessibility features such as screen reader, larger fonts, and sufficient contrast have been added into Tizen 7.0.

Reviewer: ```suggestion
  New accessibility features such as screen reader, larger fonts, and sufficient contrast have been added to Tizen 7.0.
```

>   In addition, Flutter-WebRTC was introduced to capture audio/videos and communicate using popular WebRTC framework on the Flutter Tizen apps.

Reviewer: ```suggestion
  In addition, Flutter-WebRTC was introduced to capture audio/videos and communicate using the popular WebRTC framework on Flutter Tizen apps.
```

  But we thought this was not enough to fully access the powerful features of Tizen, so we introduced native platform API bindings in Tizen 7.0.
  It provides a way to call any public Tizen native C and C# APIs from Flutter Tizen applications.

  **Figure: Native Platform API Binding Support**

  ![NativePlatformAPIBindingSupport](./media/7.0_NativePlatformAPIBindingSupport.png)

- **.NET Multi-platform App UI (.NET MAUI) for Tizen**:

>   The next advancement is the new Maui support on Tizen.Maui is the latest cross platform framework by Microsoft’s announced this past May.MAUI apps will be supported by Tizen 7.0.

Reviewer: ```suggestion
  The next advancement is the new MAUI support on Tizen. MAUI is the latest cross platform framework by Microsoft’s announced this past May. MAUI apps will be supported by Tizen 7.0.
```

  Based on the widely used .NET framework and Visual Studio IDE, Maui has three key improvements over the previous version, Xamarin.

>   First single project, Maui provides a single project for Android, iOS, Windows and Tizen. Much simpler and cleaner than before.

Reviewer: ```suggestion
  First is single project, where MAUI provides a single project for Android, iOS, Windows and Tizen. This makes it much simpler and cleaner than before.
```

>   Second, is the rich UI controls and easy UI customization. Third, essential device APIs are officially added into MAUI.With .NET MAUI,

Reviewer: ```suggestion
  Second is the rich UI controls and easy UI customization. Third, essential device APIs are officially added into MAUI. With .NET MAUI,
```

  you can take also advantage of Blazor Hybrid which provides a way to create interactive web UI with C# APIs.

  **Figure: .NET Multi-platform App UI (.NET MAUI) for Tizen**

  ![.NETMultiPlatformAppUI](./media/7.0_.NETMultiPlatformAppUI.png)
  
## Tizen as IoT Platform

- **Matter protocol for universal Smart Home ecosystem**:

>   Matter uses thread and WiFi for data communication and Bluetooth is used for provisioning.Through Matter certification,

Reviewer: ```suggestion
  Matter uses thread and WiFi for data communication and Bluetooth is used for provisioning. Through Matter certification,
```

>   Matter devices can trust each other and guarantee correctness of behavior with other matter devices.

Reviewer: ```suggestion
  Matter devices can trust each other and guarantee the correctness of behavior with other Matter devices.
```

  The Tizen team has actively participated in the Matter open source project and the first version of open source Matter SDK is now ready to use.

>   Tizen TV already passed all the tests of Matter specification validation event and is now ready to be Matter certified.

Reviewer: ```suggestion
  Tizen TV already passed all the tests of the Matter specification validation event and is now ready to be Matter certified.
```

# Tizen 7.0

The second milestone (M2) release of Tizen 7.0 was announced in October 2022. 
Tizen is an open-source operating system (OS) maintained by Samsung, and it is being developed and optimized for smart devices since 2012.

> The main progress of Tizen 7.0 includes work that has been done to accommodate new Tizen devices and strengthening intelligent device and service platform features. Core platform technologies are needed to prepare for our future with the Metaverse and robots.

Reviewer: ```suggestion
The main progress of Tizen 7.0 includes work that has been done to accommodate new Tizen devices and strengthen intelligent device and service platform features. Core platform technologies are needed to prepare for our future with the Metaverse and robots.
```

These new enhancements include 3D rendering engine enhancement, real-time enabled kernel, and more.

## Natural user interface

- **Mini Avatar UI Assistant(Consistent Interaction Through Voice)**:

  Tizen provides UI assistance technology that allows users to control the UI through voice without additional processing in applications

  The Mini Avatar is displayed in a separate layer, and it enables more natural interaction by showing the appropriate feedback for the situation.
  It also shows automatically generated commands for all controllable UI objects, and when the user says a command, it sends events to a selected object.

  **Figure: Mini Avatar UI Assistant**

  ![MiniAvartarUIAssistance](./media/7.0_MiniAvartarUIAssistance.png)

- **3D User Interface**:

  Tizen provides a unified rendering engine. All UI elements have depth information and they are located in a 3D space.

  As all UI elements are rendered together, we can optimize the rendering pass and minimize memory usage.

	The biggest advantage is that it reduces the complexity quite significantly for creating a 3D UI.

> 	This is because almost all the features used in 2D UI applications such as event handling, text input, animations, and accessibility, etc., are still available in 3D

Reviewer: ```suggestion
	This is because almost all the features used in 2D UI applications such as event handling, text input, animations, accessibility, etc., are still available in 3D
```

  **Figure: 3D Scene Building**

  ![3DSceneBuilding](./media/7.0_3DSceneBuilding.png)

	**Figure: 3D Scene Navigation**

  ![3DSceneNavigation](./media/7.0_3DSceneNavigation.png)

## Platform Compatibility

- **Tizen-Android HAL Adaptation**:

>   Lots of time and effort are used to develop hardware abstraction layers and Tizen has introduced HAL APIs in Tizen 6.5.

Reviewer: ```suggestion
  Lots of time and effort are used to develop hardware abstraction layers, so Tizen has introduced HAL APIs in Tizen 6.5.
```

> 	While we had some success with this, many SoC chip makers only support Android. To address this issue,

Reviewer: ```suggestion
	While we had some success with this, many SoC chipmakers only support Android. To address this issue,
```

	we introduced a new solution to run Tizen on Android supported SoC by providing an adaptation bridge to Android HAL
	
	To make Tizen run on Android HAL and Kernel, we added the open source project Libhybris and defined a new glue layer to access hardware devices through Android HAL.
  Also, to solve compatibility issues between bionic in Android and glibc in Tizen, we used separate namespaces for the Android System and Hal container.
  
  **Figure: Tizen-Android Hal Adaptation**

  ![TizenAndroidHalAdaptation](./media/7.0_TizenAndroidHalAdaptation.png)

## Cross Platform Technology
	
[!NOTE]
If you are already using popular cross-platform frameworks such as Web, Flutter or Maui, you can deploy your services to Tizen with minimal effort
	
- **New support for Tizen Web eco platform**:

  Web is arguably the most widely used cross-platform technology. And more and more services are based on web on Tizen devices,
  so it is business critical to support up-to-date web engines. To achieve that, we had to reduce time and effort to port a new version of the open source Chromium web engine.

>   A big move we took with Tizen 7.0 was re-architecturing our web engine porting layer. Instead of using our thick Tizen platform port,

Reviewer: ```suggestion
  A big move we took with Tizen 7.0 was re-architecting our web engine porting layer. Instead of using our thick Tizen platform port,
```

>   we now use Chromium’s aura windows system and ozone abstraction layer. As a result, it reduced significant porting effort, around 50% of code was removed,

Reviewer: ```suggestion
  we now use Chromium’s Aura window system and Ozone abstraction layer. As a result, it reduced significant porting effort, around 50% of the code was removed,
```

>   we now use Chromium’s Aura windows system and Ozone abstraction layer. As a result, it reduced significant porting effort, around 50% of the code was removed,

Reviewer: ```suggestion
  we now use Chromium’s Aura window system and Ozone abstraction layer. As a result, it reduced significant porting effort, around 50% of the code was removed,
```

>   as in LOC, more than 100,000 lines total.

Reviewer: ```suggestion
  as in LOC, which more than 100,000 in lines total.
```

>   as in LOC, which more than 100,000 lines in total.

Reviewer: ```suggestion
  as in LOC, which is more than 100,000 lines in total.
```

>   as in LOC, which more than 100,000 in lines total.

Reviewer: ```suggestion
  as in LOC, which is more than 100,000 lines in total.
```

  **Figure: New support for Tizen Web eco platform**

  ![NewSupportForTizenWebEcoPlatform](./media/7.0_NewSupportForTizenWebEcoPlatform.png)

- **Flutter Support Improvement**:

  Flutter was introduced in 2016 and has gained huge popularity among developers with more than 600,000 apps in the Google Play Store today.

>   Last year, we introduced Flutter Tizen SDK at here SDC 2021, and in Tizen 7.0, we continue to support more and more first party plugins such as Google Maps,

Reviewer: ```suggestion
  Last year, we introduced Flutter Tizen SDK at SDC 2021, and in Tizen 7.0, we continue to support more and more first party plugins such as Google Maps,
```

>   and third-party plugins such as audio players, and also Tizen dedicated Flutter plugins for Tizen specific feature access. Also, you can now use Google sign-in on Flutter Tizen TVs apps.

Reviewer: ```suggestion
  and third-party plugins such as audio players, and also Tizen dedicated Flutter plugins for Tizen specific feature access. Also, you can now use Google sign-in on Flutter Tizen TV apps.
```

>   and third-party plugins such as audio players, and also Tizen dedicated Flutter plugins for Tizen specific feature access. Also, you can now use Google sign-in on Flutter Tizen TV apps.

Reviewer: ```suggestion
  third-party plugins such as audio players, and Tizen dedicated Flutter plugins for Tizen specific feature access. Also, you can now use Google sign-in on Flutter Tizen TV apps.
```

>   New accessibility features such as screen reader, larger fonts, and sufficient contrast have been added to Tizen 7.0.

Reviewer: ```suggestion
  New accessibility features, such as screen reader, larger fonts, and sufficient contrast, have been added to Tizen 7.0.
```

>   New accessibility features such as screen reader, larger fonts, and sufficient contrast, have been added to Tizen 7.0.

Reviewer: ```suggestion
  New accessibility features such as a screen reader, larger fonts, and sufficient contrast, have been added to Tizen 7.0.
```

>   In addition, Flutter-WebRTC was introduced to capture audio/videos and communicate using the popular WebRTC framework on Flutter Tizen apps.

Reviewer: ```suggestion
  In addition, Flutter-WebRTC was introduced to capture audio/video and communicate using the popular WebRTC framework on Flutter Tizen apps.
```

  But we thought this was not enough to fully access the powerful features of Tizen, so we introduced native platform API bindings in Tizen 7.0.
  It provides a way to call any public Tizen native C and C# APIs from Flutter Tizen applications.

  **Figure: Native Platform API Binding Support**

  ![NativePlatformAPIBindingSupport](./media/7.0_NativePlatformAPIBindingSupport.png)

- **.NET Multi-platform App UI (.NET MAUI) for Tizen**:

>   The next advancement is the new MAUI support on Tizen. MAUI is the latest cross platform framework by Microsoft’s announced this past May. MAUI apps will be supported by Tizen 7.0.

Reviewer: ```suggestion
  The next advancement is the new MAUI support on Tizen. MAUI is the latest cross-platform framework by Microsoft, announced this past May. MAUI apps will be supported by Tizen 7.0.
```

>   Based on the widely used .NET framework and Visual Studio IDE, Maui has three key improvements over the previous version, Xamarin.

Reviewer: ```suggestion
  Based on the widely used .NET framework and Visual Studio IDE, MAUI has three key improvements over the previous version, Xamarin.
```

>   First is single project, where MAUI provides a single project for Android, iOS, Windows and Tizen. This makes it much simpler and cleaner than before.

Reviewer: ```suggestion
  First is single project, where MAUI provides a single project for Android, iOS, Windows, and Tizen. This makes it much simpler and cleaner than before.
```

>   Second is the rich UI controls and easy UI customization. Third, essential device APIs are officially added into MAUI. With .NET MAUI,

Reviewer: ```suggestion
  Second is the rich UI controls and easy UI customization. Third, essential device APIs are officially added to MAUI. With .NET MAUI,
```

>   you can take also advantage of Blazor Hybrid which provides a way to create interactive web UI with C# APIs.

Reviewer: ```suggestion
  you can take also advantage of Blazor Hybrid, which provides a way to create interactive web UI with C# APIs.
```

>   you can take also advantage of Blazor Hybrid, which provides a way to create interactive web UI with C# APIs.

Reviewer: ```suggestion
  you can also take advantage of Blazor Hybrid, which provides a way to create interactive web UI with C# APIs.
```

  **Figure: .NET Multi-platform App UI (.NET MAUI) for Tizen**

>   ![.NETMultiPlatformAppUI](./media/7.0_.NETMultiPlatformAppUI.png)

Reviewer: Yes. This is a good thing you have pointed out. Thank you, @mijin85cho!

@dclim7 Please address this issue as well.

  
## Tizen as IoT Platform

- **Matter protocol for universal Smart Home ecosystem**:

  Matter uses thread and WiFi for data communication and Bluetooth is used for provisioning. Through Matter certification,
  Matter devices can trust each other and guarantee the correctness of behavior with other Matter devices.
  The Tizen team has actively participated in the Matter open source project and the first version of open source Matter SDK is now ready to use.
  Tizen TV already passed all the tests of the Matter specification validation event and is now ready to be Matter certified.

  **Figure: Matter**

  ![Matter](./media/7.0_Matter.png)
  

> For more information, see [Tizen 7.0 M2 release note](../../release-notes/tizen-7-0-m2.md).

Reviewer: ```suggestion
For more information, see [Tizen 7.0 M2 Release Note](../../release-notes/tizen-7-0-m2.md).
```
